### PR TITLE
process data to grab only JPS and PNG, disregard the small trash files

### PR DESCRIPTION
**Pull request by dokluch**
Previous was expired

### CODE REVIEW

**Comment by dokluch** [comment]
Updated the way ns-process-data is grabbing the images. Some hidden images may appear in the image folder along with crmeta.db file from Reality Capture. Better to disregard that straight away.

File: scripts/process_data.py [hunk 1]
```diff
@@ -156,16 +156,19 @@ def copy_images(data: Path, image_dir: Path, verbose) -> int:
     Returns:
         The number of images copied.
     """
-    image_paths = sorted(data.glob("*"))
-    idx = 0
+    allowed_exts = ['.jpg', '.png', '.tif', '.tiff']
```
Comment (tancik) [inline scripts/process_data.py:159]: Can you add `jpeg`

File: scripts/process_data.py [hunk 2]
```diff
@@ -420,51 +423,56 @@ def main(
...
             num_frames = copy_images(data, image_dir=image_dir, verbose=verbose)
-        CONSOLE.log("[bold green]:tada: Done copying images.")
-        summary_log.append(f"Starting with {num_frames} images")
 
-    if num_downscales > 0:
-        if not verbose:
-            with CONSOLE.status("[bold yellow]Downscaling images...", spinner="growVertical"):
-                downscale_images(image_dir, num_downscales, verbose=verbose)
+        if num_frames == 0:
+            CONSOLE.log("[bold red]:tada: No usable images in the data folder.")
```
Comment (tancik) [inline scripts/process_data.py:428]: haha, I don't thin this deserves a :tada:

**Review by tancik — COMMENTED** [review]
Thanks for the improvements!

**Comment by tancik** [comment]
To run the core checks locally, call `ns-dev-train`. This will also fix the broken test above.

File: scripts/process_data.py (hunk 1 above)
Comment (dokluch) [reply scripts/process_data.py:159]: Ah, sure. Forgot it's possible.

**Review by ethanweber — APPROVED** [review]
Looks good to merge! I only made a few changes. Thanks for the PR.

**Comment by dokluch** [comment]
Added fix for 0th frame

File: scripts/process_data.py [hunk 3]
```diff
@@ -420,51 +427,74 @@ def main(
...
             num_frames = copy_images(data, image_dir=image_dir, verbose=verbose)
-        CONSOLE.log("[bold green]:tada: Done copying images.")
-        summary_log.append(f"Starting with {num_frames} images")
 
-    if num_downscales > 0:
-        if not verbose:
-            with CONSOLE.status("[bold yellow]Downscaling images...", spinner="growVertical"):
-                downscale_images(image_dir, num_downscales, verbose=verbose)
+        if num_frames == 0:
+            CONSOLE.log("[bold red]No usable images in the data folder.")
```
Comment (ethanweber) [inline scripts/process_data.py:432]: If there are no frames, I think the code should exit right after. Right?

File: scripts/process_data.py [hunk 4]
```diff
@@ -420,51 +427,74 @@ def main(
...
+            # To improve the solve speed
 
-    if (colmap_dir / "sparse" / "0" / "cameras.bin").exists():
-        with CONSOLE.status("[bold yellow]Saving results to transforms.json", spinner="balloon"):
-            num_matched_frames = colmap_to_json(
-                cameras_path=colmap_dir / "sparse" / "0" / "cameras.bin",
-                images_path=colmap_dir / "sparse" / "0" / "images.bin",
-                output_dir=output_dir,
+            max_side = max(Image.open(next(image_dir.glob("[!.]*"))).size)
+            factor = max_side // 2000
```
Comment (ethanweber) [inline scripts/process_data.py:444]: Should 2000 be a parameter someone can change from the terminal? If so, we can document it there.

**Review by ethanweber — COMMENTED** [review]
Made a few comments. Thanks for these changes.

**Comment by ethanweber** [comment]
Can you also update the name of the PR? We should merge it after you address the minor comments just to get it in. Thanks!

File: scripts/process_data.py (hunk 4 above)
Comment (dokluch) [reply scripts/process_data.py:444]: Sure. I forgot this PR is still opened and pushed without consideration. I am still doing tests so I'd rather roll back a commit

**Comment by ethanweber** [comment]
Closing this PR and replacing with PR #756. @dokluch you can make another PR when ready for new changes.

**Pull request closed without merging**